src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -172,8 +172,16 @@ struct HostFunctionMetrics
     uint32 mEmitEvent{0};
     uint32 mEmitEventByte{0};
 
+    // host runtime metrics
     uint32 mCpuInsn{0};
     uint32 mMemByte{0};
+    uint32 mInvokeTimeNsecs{0};
```

Comment:
Shouldn't this be a uint64?

Reply:
I noticed @MonsieurNicolas recently changed all metrics to `uint32` and explicitly casting all values to u32 https://github.com/stellar/stellar-core/pull/3833/commits/5ee769ff56590b6b0365cd9b02496e13b111fb74#diff-12e712ade191c619e26f360d118226148677a891b5d23f2474ebe64c427351e9R157-R178. 
I think this should be okay too since u32::MAX nsecs is more than 4 secs which should be higher than the practical upper bound.

Reply:
Uh I'm not sure about that -- @MonsieurNicolas why did you insert all those uint32 casts? A lot of them look .. fairly aggressive and maybe risk losing information. Does win32 not like uint64 for some reason?